test/vfd_swmr_bigset_writer.c

```diff
@@ -875,64 +1381,99 @@ verify_dset_attribute(hid_t ds, unsigned int which, unsigned int step)
-        errx(EXIT_FAILURE, "H5Aclose failed");
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Aclose(aid);
+    }
+    H5E_END_TRY;
 
-    if (read_step != step)
-        errx(EXIT_FAILURE, "expected %u read %u", step, read_step);
+    return false;
 }
 
-static void
-verify_extensible_dset(state_t *s, unsigned int which, mat_t *mat, unsigned *stepp)
+static bool
+verify_extensible_dset(state_t *s, unsigned int which, mat_t *mat, unsigned finished_step, unsigned last_step)
 {
-    hid_t        ds, filespace;
+    hid_t        dset_id = H5I_INVALID_HID, filespace = H5I_INVALID_HID;
     hsize_t      size[RANK];
     base_t       base, last;
-    unsigned int ncols, last_step, step;
-
-    assert(which < s->ndatasets);
+    unsigned int ncols, step;
+    int          i;
 
-    ds = s->dataset[which];
+    if (which >= s->ndatasets) {
+        fprintf(stderr, "the dataset order is bigger than the number of datasets");
+        TEST_ERROR;
+    }
 
-    if (H5Drefresh(ds) < 0)
-        errx(EXIT_FAILURE, "H5Drefresh failed");
+    dset_id = s->dataset[which];
 
-    filespace = H5Dget_space(ds);
+    /* Attempt to check the availablity of the chunks for a number time before reporting it as a failure */
+    for (i = 0; i < NUM_ATTEMPTS; i++) {
+        if (H5Drefresh(dset_id) < 0) {
+            fprintf(stderr, "H5Drefresh failed\n");
+            TEST_ERROR;
+        }
 
-    if (filespace == badhid)
-        errx(EXIT_FAILURE, "H5Dget_space failed");
+        if ((filespace = H5Dget_space(dset_id)) < 0) {
+            fprintf(stderr, "H5Dget_space failed\n");
+            TEST_ERROR;
+        }
 
-    if (H5Sget_simple_extent_dims(filespace, size, NULL) < 0)
-        errx(EXIT_FAILURE, "H5Sget_simple_extent_dims failed");
+        if (H5Sget_simple_extent_dims(filespace, size, NULL) < 0) {
+            fprintf(stderr, "H5Sget_simple_extent_dims failed\n");
+            TEST_ERROR;
+        }
 
-    ncols = (unsigned)(size[1] / s->chunk_dims[1]);
-    if (ncols < hang_back)
-        goto out;
+        ncols = (unsigned)(size[1] / s->chunk_dims[1]);
 
-    last_step = ncols - hang_back;
+        /* Make sure the chunks show up on the reader side.  Otherwise sleep a while and try again */
+        if (ncols >= last_step)
+            break;
+        else
+            decisleep(1);
+    }
 
-    for (step = *stepp; step <= last_step; step++) {
-        const unsigned ofs = step % 2;
+    if (i == NUM_ATTEMPTS) {
+        fprintf(stderr, "chunk verification reached the maximal number of attempts");
+        TEST_ERROR;
+    }
 
+    for (step = finished_step; step < last_step; step++) {
         dbgf(1, "%s: which %u step %u\n", __func__, which, step);
 
         if (s->two_dee) {
             size[0]  = s->chunk_dims[0] * (1 + step);
             size[1]  = s->chunk_dims[1] * (1 + step);
-            last.row = s->chunk_dims[0] * step + ofs;
-            last.col = s->chunk_dims[1] * step + ofs;
+            last.row = s->chunk_dims[0] * step;
```

Comment:
Would you please explain a bit why "+ofs" is no longer needed?  Thank you.

Reply:
I put it back.